<?xml version="1.0" encoding="UTF-8"?>
<launch>
    <arg name="enable_rviz" default="true" />
    <group if="$(arg enable_rviz)">
        <node pkg="rviz" type="rviz" name="sloam_rviz" output="screen" args="-d $(find sloam)/launch/rviz/decentralized_sloam.rviz" />
    </group>

    <rosparam file="$(find sloam)/params/sloam.yaml" command="load" />
    <arg name="hostRobot0_ID" default="0" />
    <arg name="turn_off_intra_loop_closure" default="true"/>

    <node pkg="sloam" type="sloam_node" name="sloam0" output="screen">
            <param name="hostRobotID" value="$(arg hostRobot0_ID)" />
            <!-- the TF here is from robot to world -->
            <param name="turn_off_intra_loop_closure" value="$(arg turn_off_intra_loop_closure)"/>
            <param name="priorTFKnown" value="false"/>
            <param name="priorTF_x" value="0"/>
            <param name="priorTF_y" value="0"/>
            <param name="priorTF_z" value="0"/>
            <!-- for publish -->
             <remap from="/sloam/cubes_map"                  to="robot_$(arg hostRobot0_ID)/sloam/cubes_map"/>
            <remap from="/sloam/cubes_submap"                to="robot_$(arg hostRobot0_ID)/sloam/cubes_submap"/>
            <remap from="/sloam/debug/robot0/trajectory"     to="robot_$(arg hostRobot0_ID)/sloam/debug/robot0/trajectory"/>
            <remap from="/sloam/debug/robot1/trajectory"     to="robot_$(arg hostRobot0_ID)/sloam/debug/robot1/trajectory"/>
            <remap from="/sloam/cylinders_map"                         to="robot_$(arg hostRobot0_ID)/sloam/cylinders_map"/>
            <remap from="/sloam/submap_cylinder_models"         to="robot_$(arg hostRobot0_ID)/sloam/submap_cylinder_models"/>
            <remap from="/sloam/map_pose"                    to="robot_$(arg hostRobot0_ID)/sloam/map_pose"/>
            <remap from="/sloam/observation"                 to="robot_$(arg hostRobot0_ID)/sloam/observation"/>
            <remap from="/sloam/optimized_point_landmarks"         to="robot_$(arg hostRobot0_ID)/sloam/optimized_point_landmarks"/>
            <!-- from desired to actual (original) topic -->
            <!-- <remap from="/quadrotor0/semantic_meas_sync_odom"     to="/robot_0/semantic_meas_sync_odom"/> -->
            <!-- <remap from="/quadrotor0/semantic_meas_sync_odom"     to="/semantic_meas_sync_odom"/> -->
            <!-- <remap from="/sloam/odom"         to="/quadrotor1/lidar_odom"/> -->
            <!-- <remap from="/sloam/odom"         to="/robot_0/vio_odom"/> -->
            <!-- <remap from="/sloam/odom"         to="/quadrotor/lidar_odom"/> -->
            <remap from="/sloam/odom"         to="/Odometry"/>
    </node>

</launch>
